### PR TITLE
[VPC][SubnetPort] Update attachment_ref annotation format

### PR DESCRIPTION
**Pull request by gran-vmv**
Follow the change in https://github.com/vmware-tanzu/vm-operator/pull/624

### CODE REVIEW

File: pkg/controllers/common/utils.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func GetVirtualMachineNameForSubnetPort(subnetPort *v1alpha1.SubnetPort) (string
 		return "", nil
 	}
 	array := strings.Split(attachmentRef, "/")
-	if len(array) != 2 || !strings.EqualFold(array[0], servicecommon.ResourceTypeVirtualMachine) {
+	if len(array) != 3 || !strings.EqualFold(array[0], servicecommon.ResourceTypeVirtualMachine) {
 		err := fmt.Errorf("invalid annotation value of '%s': %s", servicecommon.AnnotationAttachmentRef, attachmentRef)
 		return "", err
 	}
```
Comment (dantingl) [inline pkg/controllers/common/utils.go:137]: Will you have another patch to put interface name in store?

Comment (gran-vmv) [reply pkg/controllers/common/utils.go:137]: This PR is to unblock integration test. Other changes will be added to controller PR.

Comment (gran-vmv) [reply pkg/controllers/common/utils.go:137]: Currently nsx-operator doesn't have extra indexer for K8s store. I checked vmMapFunc and it listed all SubnetPorts to find proper VM name, and I'm not sure how to add K8s store indexer into current nsx-operator.

**Review by dantingl — APPROVED** [review]

**Review by heypnus — APPROVED** [review]

**Pull request merged**